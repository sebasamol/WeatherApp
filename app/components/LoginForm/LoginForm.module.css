.container {
    position: absolute;
    top: 25%;
    left: 50%;
    transform: translate(-50%, -50%);
    height: 400px;
    width: 400px;

}

.description {
    width: 350px;
    margin: auto;
    display: flex;
    justify-content: flex-start;
    align-items: center;

}

.description p {
    margin-left: 15px;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 16px;
    letter-spacing: 2px;
    word-spacing: -1px;
    color: #ffff;
    font-weight: normal;
    text-decoration: none;
    font-style: normal;
    font-variant: normal;
    text-transform: uppercase;
}

.form_container {
    margin: auto;
    width: 400px;
    margin-top: 35px;
}

.form_data {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    gap: 8px;

}

/* components */
.input_group {
    position: relative;

}

.input_group input {
    width: var(--input-width);
    height: var(--input-height);
    padding: 10px;
    border: 1px solid rgba(255, 255, 255, 0.597);
    border-radius: var(--border-radius);
    background: transparent;
    outline: none;
    color: #fff;

}

.input_group label {
    position: absolute;
    left: 5px;
    top: 18px;
    pointer-events: none;
    text-transform: uppercase;
    transition: 0.5s;
    font-size: 10px;
    color: rgba(255, 255, 255, 0.565);

}

.input_group input:focus~label,
.input_group input:valid~label {
    transform: translateX(5px) translateY(-23px);
    font-size: 0.55em;
    padding: 0 10px;
    background: #1b2b3a;
    border-left: 1px solid white;
    border-right: 1px solid white;
    color: rgb(1, 247, 255);
    letter-spacing: 0.2em;

}


.button {
    height: var(--button-height);
    width: var(--button-width);
    border-radius: var(--border-radius);
    border: 1px solid rgba(255, 255, 255, 0.597);
    box-sizing: border-box;
    background: transparent;
    color: white;
    margin-top: 5px;
    font-size: 18px;

}

.button:hover {
    background: var(--form-hover);
    transition: 0.6s;
    opacity: 0.85;


}