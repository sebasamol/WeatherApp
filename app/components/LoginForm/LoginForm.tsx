'use client'
import React, { useState } from "react";
import { FormEvent } from "react";
import { useRouter } from 'next/navigation';
import Image from "next/image";
import styles from './LoginForm.module.css'


export default function LoginForm() {
    const router = useRouter()
    const [isLoading, setIsLoading] = useState<boolean>(false)
    const [error, setError] = useState<string | null>(null)
    const [username, setUsername] = useState("")
    const [password, setPassword] = useState("")

    async function handleSubmit(event: FormEvent<HTMLFormElement>) {

        event.preventDefault()
        setIsLoading(true)
        setError(null)

        try {
            const formData = new FormData(event.currentTarget)
            const response = await fetch('/pages/api/submit', {
                method: 'POST',
                body: formData,
            })

            if (response.status == 200) {
                router.push('/')
                
            } else {
                setError('Nieprawidłowa nazwa użytkownika lub hasło')

            }

        } catch (error) {
            console.log(error)
            
        } finally {
            setIsLoading(false)
        }

    }
    return (
        <div className={styles.container}>
            <main className={styles.main}>
                <div className={styles.description}>
                    <Image
                        src={'/weather-station.png'}
                        width={64}
                        height={64}
                        alt="Weather station"
                    />
                    <p>Strona logowania</p>
                </div>
                <div className={styles.form_container}>
                    <form className={styles.form_data} onSubmit={handleSubmit} aria-label="signup-form">
                        <div className={styles.input_group}>
                            <input
                                className={styles.input}
                                id='login'
                                name='login'
                                type='text'
                                data-testid="test-username"
                                required={true}
                                defaultValue={username}
                                onChange={(e) => setUsername(e.target.defaultValue)}
                                
                            ></input>
                            <label className={styles.label}>Nazwa użytkownika</label>

                        </div>
                        <div className={styles.input_group}>
                            <input
                                className={styles.input}
                                id='password'
                                name='password'
                                type='password'
                                data-testid="test-password"
                                required={true}
                                defaultValue={password}
                                onChange={(e) => setPassword(e.target.defaultValue)}
                            ></input>
                            <label className={styles.label}>Hasło</label>

                        </div>

                        {error && <div className={styles.display_error} >
                            <Image
                                src="/error.png"
                                width={18}
                                height={18}
                                alt="Error"

                            />
                            <div data-testid="error" className={styles.error} >{error}</div></div>}

                        <button className={styles.button} data-testid='test-button'type="submit" disabled={isLoading}>
                            {isLoading ? 'Logowanie...' : 'Zaloguj'}</button>
                    </form>

                </div>
            </main>

        </div>

    )
}

