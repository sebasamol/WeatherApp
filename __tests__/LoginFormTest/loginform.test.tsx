import "@testing-library/jest-dom"
import { render, screen } from '@testing-library/react'
import LoginForm from "@/app/components/LoginForm/LoginForm"
import exp from "constants";

jest.mock("next/navigation", () => ({
    useRouter() {
      return {
        prefetch: () => null
      };
    }
  }));
  
describe("Login Form test", () => {
    test ('should take a snapshot', () =>{
      const { asFragment } = render(<LoginForm/>)
      expect(asFragment()).toMatchSnapshot()
    })
    
    test('username input should be rendered', () => {
      render(<LoginForm/>)
      const userInputEl = screen.getByText('Nazwa użytkownika')
      expect(userInputEl).toBeInTheDocument()
    })
    test('password input should be rendered', () => {
      render(<LoginForm/>)
      const userInputEl = screen.getByText('Hasło')
      expect(userInputEl).toBeInTheDocument()
    })
    test('login button should be rendered', () => {
      render(<LoginForm/>)
      const userInputEl = screen.getByText('Zaloguj')
      expect(userInputEl).toBeInTheDocument()
    })
    
})