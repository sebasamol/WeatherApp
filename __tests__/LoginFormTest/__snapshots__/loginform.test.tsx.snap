// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form test should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <div
        class="description"
      >
        <img
          alt="Weather station"
          data-nimg="1"
          decoding="async"
          height="64"
          loading="lazy"
          src="/_next/image?url=%2Fweather-station.png&w=128&q=75"
          srcset="/_next/image?url=%2Fweather-station.png&w=64&q=75 1x, /_next/image?url=%2Fweather-station.png&w=128&q=75 2x"
          style="color: transparent;"
          width="64"
        />
        <p>
          Strona logowania
        </p>
      </div>
      <div
        class="form_container"
      >
        <form
          class="form_data"
        >
          <div
            class="input_group"
          >
            <input
              class="input"
              name="login"
              placeholder=""
              required=""
              type="text"
            />
            <label
              class="label"
            >
              Nazwa użytkownika
            </label>
          </div>
          <div
            class="input_group"
          >
            <input
              class="input"
              name="password"
              placeholder=""
              required=""
              type="password"
            />
            <label
              class="label"
            >
              Hasło
            </label>
          </div>
          <button
            class="button"
            type="submit"
          >
            Zaloguj
          </button>
        </form>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
